### PR TITLE
feat: dislike blog post

### PR DESCRIPTION
**Pull request by traderstechie**
## Description:
A feature that allows users to react to a blog post with a dislike. It's a protected endpoint at `/api/v1/blogs/:{blog_id}/dislike`. It allows only authenticated users to dislike a blog post once.

## Related Issue:

[[FEAT]: Endpoint To Dislike A Blog Post #275](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/275)

## Motivation and Context
This is necessary, in order to enable blog writers to understand how readers perceive their posts.

## How Has This Been Tested?

- [x] Unittest with pytest.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:
 
- [x]  My code follows the code style of this project.
- [x]  I have updated the documentation accordingly.
- [x]  I have read the CONTRIBUTING document.
- [x]   I have added tests to cover my changes.
- [x]  All new and existing tests passed.

### CODE REVIEW

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

File: api/v1/routes/blog.py [hunk 1]
```diff
@@ -83,6 +85,54 @@ async def update_blog(id: str, blogPost: BlogRequest, db: Session = Depends(get_
...
+    if not current_user:
+        return JsonResponseDict(
+            message="Could not validate credentials",
+            status_code=status.HTTP_401_UNAUTHORIZED
+        )
+
+    # GET blog post
+    blog_p = blog_service.fetch(blog_id)
+    if not isinstance(blog_p, Blog):
+        return JsonResponseDict(
```
[the remark this answers is not in the document]
Comment (traderstechie) [reply api/v1/routes/blog.py:108]: Okay

**Comment by traderstechie** [comment]
Changes made @joboy-dev

File: api/v1/routes/blog.py
Comment (joboy-dev) [inline api/v1/routes/blog.py:1]: Do not return success_response for errors please. Just raise HTTPException

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes

**Review by Goketech — APPROVED** [review]

**Review by joboy-dev — APPROVED** [review]

**Pull request merged**